include/dali/core/util.h

```diff
@@ -143,6 +143,52 @@ constexpr std::enable_if_t<std::is_integral<T>::value, int> ilog2(T x) {
   return n;
 }
 
+
+/**
+ * @brief Count Trailing Zeros (CTZ)
+ *
+ * @return The number of trailing (LSB) zeros or the number of bits in the argument,
+ *         if it's equal to 0.
+ */
+template <typename Integer>
+DALI_HOST_DEV DALI_FORCEINLINE
+std::enable_if_t<std::is_integral<Integer>::value, int> ctz(Integer word) {
+  std::make_unsigned_t<Integer> uword = word;
+  if (uword == 0)
+    return sizeof(uword) * 8;
+  // The hardware can check for multiple bits at a time, so we don't have to scan sequentially
+  int bit = 0;
+  if (sizeof(uword) > 4 && (uword & 0xffffffffu) == 0) {
+    bit += 32;
+    const int shift32 = sizeof(uword) > 4 ? 32 : 0;  // workaround undefined shift warning
+    uword >>= shift32;
```

Comment:
```suggestion
    const int shift32 = 32;  // workaround undefined shift warning
    uword >>= shift32;
```
The above if already guarantees that.

Reply:
Or this is just for the warning?

Reply:
Yes, it's for the compiler warning.